lib/OpenQA/Task/Job/ArchiveResults.pm

```diff
@@ -0,0 +1,43 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Task::Job::ArchiveResults;
+use Mojo::Base 'Mojolicious::Plugin', -signatures;
+
+sub register ($self, $app, @args) {
+    $app->minion->add_task(archive_job_results => \&_archive_results);
+}
+
+sub _archive_results ($minion_job, @args) {
+    my ($openqa_job_id) = @args;
+    my $app = $minion_job->app;
+    return $minion_job->fail('No job ID specified.') unless defined $openqa_job_id;
+
+    # FIXME: Should we ensure that the job results are finalized?
+    # FIXME: Should we take care of possible interference with cleanup task?
```

Comment:
do you mean results or screenshots cleanup? Because when triggering synchronously with results cleanup then that at least should not interfer. And screenshots are symlinks which should still be valid after archiving a job unless any screenshot cleanup happens in parallel to the directory copy, right? In that case, how about a minion lock on each corresponding job?

Reply:
> do you mean results or screenshots cleanup? Because when triggering synchronously with results cleanup then that at least should not interfer.

I mean the result cleanup. I suppose it wouldn't interfere in practice but I can think of some contrived examples, e.g. a job is no longer considered important during the cleanup and then would be deleted after all (while the archiving is ongoing).

> And screenshots are symlinks which should still be valid after archiving a job unless any screenshot cleanup happens in parallel to the directory copy, right?

Yes, screenshots are not touched by this. And the symlinks remain valid as long as they use absolute paths which they luckily do. If screenshot cleanup happens in the background that shouldn't be a problem because it wouldn't delete screenshots used by existing jobs (regardless of their archiving state).